services/resource_dereference_service.go

```diff
@@ -0,0 +1,101 @@
+package services
+
+import (
+	// jsonpb Marshaller is deprecated, but is needed because there's only one way to proto
+	// marshal in combination with our proto generator version
+	"encoding/json"
+
+	"github.com/cheqd/did-resolver/types"
+	"github.com/cheqd/did-resolver/utils"
+)
+
+type ResourceDereferenceService struct {
```

Comment:
As a general idea. What do you think about addin interfaces to all services? Just to make clearer what the method list is.

Reply:
The same way it's done for LedgerService